.github/workflows/update-docs.yml

```diff
@@ -0,0 +1,53 @@
+name: Update Template Documentation
+
+on:
+  schedule:
+    - cron: '0 16 * * 2' # Every Tuesday at 11 AM EST (16:00 UTC)
+  workflow_dispatch:
+
+jobs:
+  update-docs:
+    name: Update Template Documentation
+    runs-on: ubuntu-latest
+    if: github.repository == 'GoogleCloudPlatform/DataflowTemplates'
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Install Templates Plugin
+        run: |
+          mvn clean install -pl plugins/templates-maven-plugin -am
+      - name: Generate Documentation
+        run: |
+          mvn clean prepare-package \
+            -DskipTests \
+            -PtemplatesSpec
+
+      - name: Check for changes
+        id: changes
+        run: |
+          git status --porcelain
```

Comment:
Won't the action automatically handle this? Per https://github.com/peter-evans/create-pull-request?tab=readme-ov-file#-create-pull-request the action will "Check for repository changes in the Actions workspace"